ADR/0030-tekton-results-naming-convention.md

```diff
@@ -0,0 +1,257 @@
+    "medium": 0,
+    "low":1
+  }
+}
+```
+
+#### `SCAN_OUTPUT` Schema Validation
+The output of the Tekton result **SCAN_OUTPUT** can be validated using the following schema:
+
+```
+{
+  "$schema": "http://json-schema.org/draft/2020-12/schema#",
+  "type": "object",
+  "properties": {
+    "vulnerabilities": {
+      "type": "object",
+      "properties": {
+        "critical": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "high": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "medium": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "low": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "unknown": {
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "required": ["critical", "high", "medium", "low"]
+    },
+    "unpatched_vulnerabilities": {
+      "type": "object",
+      "properties": {
+        "critical": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "high": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "medium": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "low": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "unknown": {
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "required": ["critical", "high", "medium", "low"]
+    }
+  },
+  "required": ["vulnerabilities"]
+}
+```
+
+### Results for Tasks that Process Multiple Images
+
+When scans and tests ingest an image reference as input, the actual content that is tested or scanned is ambiguous due to the potential presence of multi-arch images. If the image reference is an Image Manifest, for example, then that image would be known to be scanned. If the image reference is an Image Index, however, the Task could be processing all images referenced (i.e. all platforms/architectures), or just a single Image Manifest such as the one corresponding the the pod's architecture.
+
+In order to disambiguate the image references corresponding to the other Tekton results, a Task may output an additional result indicating the images that have been processed in order to obtain the otherwise specified result(s). The name of the result will be **IMAGES_PROCESSED**.
+
+#### Tekton Result Format for `IMAGES_PROCESSED`
+The test output of the Tekton result **IMAGES_PROCESSED** will be a JSON object that includes the context about the image references that have been processed.
+
+In the case that the image processed is an Image Manifest, the result may look like the following:
+
+```
+{
+  "image": {
+    "pullspec": "quay.io/foo/bar:baz",
+    "digests": [
+      "@sha256:bb8[...]"
+    ]
+  }
+}
+```
+
+In the case that the image processed is an Image Index, the result may look like the following:
+
+```
+{
+  "image": {
+    "pullspec": "quay.io/foo/bar:baz",
+    "digests": [
+      "@sha256:bb8[...]",
```

Comment:
Those digests don't hold any information about architecture, so basically I've to query them in advance to detect architecture and do something, so basically we can have index digest and we can do the same query

Reply:
Maybe I just finally get it, this is not output of build task, and input of integration task, but output from integration task, so at the same time we report, SCAN_RESULT and IMAGES_PROCESSED

Reply:
Yes, that is correct. This would surface the information about what images are processed by the test tasks. The validation that all relevant digests (i.e. corresponding to the different architectures) would be done at a later point according to a policy (i.e. via Enterprise Contract). We just need to make sure that the relevant contextual information is available to the policy engine in able for a fully informed decision to be made.